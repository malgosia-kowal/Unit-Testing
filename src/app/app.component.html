<div class="header">
  <h1>
    {{ title }}
  </h1>
</div>
<div class="container">
  <app-products></app-products>
  <app-quickview></app-quickview>
  <app-basket></app-basket>
</div>