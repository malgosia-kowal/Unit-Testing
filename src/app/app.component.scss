.header {
  background: #fff;
  box-shadow: 0 1px 4px rgba(0,0,0,.15);
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 100;
}

h1 {
  color: black;
  text-align: center;
}

.container {
  margin-top: 50px;
  margin-bottom: 50px;
  background-color: #f1f1f1f1;
  height: 100vh;
}