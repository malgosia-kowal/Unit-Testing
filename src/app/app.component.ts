import { Component } from '@angular/core';
import { TranslateService } from '@ngx-translate/core';

export enum Locale {
  Pl = 'pl',
  Gb = 'en',
}

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss']
})
export class AppComponent {
  title = 'Underground Cat Store (Meow)';
  locale;

  constructor(private translate: TranslateService) {
    translate.setDefaultLang(Locale.Gb);
    this.locale = Locale;
  }

  selectLocale(locale: Locale) {
    this.translate.use(locale);
  }
}
