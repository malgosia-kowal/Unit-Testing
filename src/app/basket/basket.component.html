<div class="basketContainer">
  <div class="basketContainerDetail">
    <div> Total: {{ basketService.total }} zl</div>
    <div> Products: {{ basketService.products.length }}</div>
  </div>
  <div>
    <app-button
      text="Clean"
      (action)="basketService.clear()"
    >
    </app-button>
  </div>
</div>
