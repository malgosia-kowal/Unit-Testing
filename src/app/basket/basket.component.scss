.basketContainer {
  z-index: 999;
  position: fixed;
  bottom: 0;
  background: white;
  width: 100vw;
  box-shadow: 0 0 10px rgb(0, 0, 0, 0.3);
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  padding: 10px;

  &Detail {
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;

    & > div {
      font-size: 20px;
    }
  }

  &Buttons {
    display: flex;
    flex-direction: row;
    margin-right: 40px;
  }
}
