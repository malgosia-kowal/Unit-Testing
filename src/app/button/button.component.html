<button [ngStyle]="styles" (click)="action.emit('click')">
    {{ text }}
</button>
