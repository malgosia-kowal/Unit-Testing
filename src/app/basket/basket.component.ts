import { Component, OnInit } from '@angular/core';
import { BasketService } from '../service/basket.service';
import { Toggable, ToggleService } from '../service/toggle.service';

@Component({
  selector: 'app-basket',
  templateUrl: './basket.component.html',
  styleUrls: ['./basket.component.scss']
})
export class BasketComponent implements OnInit {

  constructor(public basketService: BasketService, public toggleService: ToggleService) { }

  ngOnInit() {
  }

  toggleQuickview() {
    this.toggleService.toggle(Toggable.Quickview);
  }

}
