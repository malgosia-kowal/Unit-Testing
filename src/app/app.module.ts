import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { TranslateModule, TranslateLoader } from '@ngx-translate/core';
import { HttpClientModule, HttpClient } from '@angular/common/http';
import { TranslateHttpLoader } from '@ngx-translate/http-loader';

import { AppRoutingModule } from './app-routing.module';
import { AppComponent } from './app.component';
import { ProductsComponent } from './products/products.component';
import { ProductDetailComponent } from './product-detail/product-detail.component';
import { BasketComponent } from './basket/basket.component';
import { ButtonComponent } from './button/button.component';
import { QuickviewComponent } from './quickview/quickview.component';

export function HttpLoaderFactory(http: HttpClient) {
  return new TranslateHttpLoader(http, 'assets/i18n/', '.json');
}

@NgModule({
  declarations: [
    AppComponent,
    ProductsComponent,
    ProductDetailComponent,
    BasketComponent,
    ButtonComponent,
    QuickviewComponent
  ],
  imports: [
    BrowserModule,
    AppRoutingModule,
    FormsModule,
    HttpClientModule,
    TranslateModule.forRoot({
      loader: {
        provide: TranslateLoader,
        useFactory: HttpLoaderFactory,
        deps: [HttpClient]
      }
    })
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
