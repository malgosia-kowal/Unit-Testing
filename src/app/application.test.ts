  import {
    async,
    ComponentFixture,
    TestBed,
  } from '@angular/core/testing';
  import { By } from '@angular/platform-browser';
  import { ProductsComponent } from './products/products.component';
  import { BasketComponent} from './basket/basket.component';
  import { QuickviewComponent } from './quickview/quickview.component';
  import { AppComponent } from './app.component';
  import { ProductDetailComponent } from './product-detail/product-detail.component';
  import { ButtonComponent } from './button/button.component';
  
  describe('App Component', () => {
    let component: AppComponent;
    let fixture: ComponentFixture<AppComponent>;

    beforeEach(async(() => {
      jest.resetAllMocks();
      TestBed.configureTestingModule({
        declarations: [AppComponent, ProductsComponent, QuickviewComponent, BasketComponent, ProductDetailComponent, ButtonComponent],
      })
        .compileComponents()
        .then(() => {
          fixture = TestBed.createComponent(AppComponent);
          component = fixture.componentInstance;
  
          fixture.detectChanges();
        });
    }));
    it('should exist', () => {
      expect(component).toBeDefined();
    });
  
    it('should take a snapchot', () => {
      expect(fixture).toMatchSnapshot();
    });

    it('has a  correct title', () => {
      expect(component.title).toContain('Underground Cat Store (Meow)');
    });

    it('contains other component tags inside', () => {
      const appComponentDetail = fixture.debugElement.query(By.css('.container'));
        
      expect(appComponentDetail).not.toBe(null);
    });  

  });