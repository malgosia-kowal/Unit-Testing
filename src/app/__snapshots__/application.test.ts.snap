// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should take a snapchot 1`] = `
<app-root
  locale={[Function Object]}
  title={[Function String]}
  translate={[Function TranslateService]}
>
  <div
    class="header"
  >
    <h1>
       Underground Cat Store (Meow) 
    </h1>
    <div
      class="flagsContainer"
    >
      <span
        class="flag-icon flag-icon-pl"
      />
      <span
        class="flag-icon flag-icon-gb"
      />
    </div>
  </div><div
    class="container"
  >
    <app-products>
      <h2>
        Products
      </h2>
      <app-product-detail>
        
      </app-product-detail>
      <div
        class="productsContainer"
      >
        
        <div
          class="productCard"
        >
          <img
            src="http://placekitten.com/250/300"
          />
          <h3>
            Pushkin
          </h3>
          <hr />
          <div>
            <span
              class="productCardPropHeader"
            >
              Color: 
            </span>
            <span
              class="productCardPropValue"
            >
              red
            </span>
          </div>
          <div>
            <span
              class="productCardPropHeader"
            >
              Price:  
            </span>
            <span
              class="productCardPropValue"
            >
              1,00
            </span>
            <span>
              gbp
            </span>
          </div>
          <hr />
          <div
            class="productCardFooter"
          >
            <app-button
              id="addToBasketButton-0"
              ng-reflect-text="Add to basket"
            >
              <button>
                 Add to basket
  
              </button>
            </app-button>
          </div>
        </div>
        <div
          class="productCard"
        >
          <img
            src="http://placekitten.com/250/300"
          />
          <h3>
            Pushkin
          </h3>
          <hr />
          <div>
            <span
              class="productCardPropHeader"
            >
              Color: 
            </span>
            <span
              class="productCardPropValue"
            >
              grey
            </span>
          </div>
          <div>
            <span
              class="productCardPropHeader"
            >
              Price:  
            </span>
            <span
              class="productCardPropValue"
            >
              1,23
            </span>
            <span>
              gbp
            </span>
          </div>
          <hr />
          <div
            class="productCardFooter"
          >
            <app-button
              id="addToBasketButton-1"
              ng-reflect-text="Add to basket"
            >
              <button>
                 Add to basket
  
              </button>
            </app-button>
          </div>
        </div>
        <div
          class="productCard"
        >
          <img
            src="http://placekitten.com/250/300"
          />
          <h3>
            Cuddle
          </h3>
          <hr />
          <div>
            <span
              class="productCardPropHeader"
            >
              Color: 
            </span>
            <span
              class="productCardPropValue"
            >
              pink
            </span>
          </div>
          <div>
            <span
              class="productCardPropHeader"
            >
              Price:  
            </span>
            <span
              class="productCardPropValue"
            >
              1,23
            </span>
            <span>
              gbp
            </span>
          </div>
          <hr />
          <div
            class="productCardFooter"
          >
            <app-button
              id="addToBasketButton-2"
              ng-reflect-text="Add to basket"
            >
              <button>
                 Add to basket
  
              </button>
            </app-button>
          </div>
        </div>
        <div
          class="productCard"
        >
          <img
            src="http://placekitten.com/250/300"
          />
          <h3>
            Charles
          </h3>
          <hr />
          <div>
            <span
              class="productCardPropHeader"
            >
              Color: 
            </span>
            <span
              class="productCardPropValue"
            >
              ginger
            </span>
          </div>
          <div>
            <span
              class="productCardPropHeader"
            >
              Price:  
            </span>
            <span
              class="productCardPropValue"
            >
              1,23
            </span>
            <span>
              gbp
            </span>
          </div>
          <hr />
          <div
            class="productCardFooter"
          >
            <app-button
              id="addToBasketButton-3"
              ng-reflect-text="Add to basket"
            >
              <button>
                 Add to basket
  
              </button>
            </app-button>
          </div>
        </div>
      </div>
    </app-products>
    <app-quickview>
      
    </app-quickview>
    <app-basket>
      <div
        class="basketContainer"
      >
        <div
          class="basketContainerDetail"
        >
          <div>
            <span
              id="basketTotal"
              ng-reflect-translate="Basket.Total"
              ng-reflect-translate-params="[object Object]"
            >
              Total: 0,00 gbp
            </span>
          </div>
          <div>
            <span
              id="basketLength"
              ng-reflect-translate="Basket.Items"
              ng-reflect-translate-params="[object Object]"
            >
              Cats: 0
            </span>
          </div>
        </div>
        <div
          class="basketContainerButtons"
        >
          <app-button
            id="clearTheBasketButton"
            ng-reflect-text="Clean"
          >
            <button>
               Clean
  
            </button>
          </app-button>
          <app-button
            id="quickViewButton"
            ng-reflect-text="Quick view"
          >
            <button>
               Quick view
  
            </button>
          </app-button>
        </div>
      </div>
    </app-basket>
  </div>
</app-root>
`;
